.github/workflows/macos-build.yml

```diff
@@ -0,0 +1,35 @@
+name: Build FStar Binaries
+
+on: [push, pull_request]
+
+jobs:
+
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - name: Checkout everest
+      uses: actions/checkout@v2
+      with:
+        repository: project-everest/everest
+        path: everest
+    - name: Checkout FStar
+      uses: actions/checkout@v2
+      with:
+        path: FStar
+    - name: Setup dependencies
+      run: ./FStar/.github/setup.sh
+    - name: Build FStar
+      run: |
+        source FStar/.github/env.sh
+        make -j -C FStar all
+    - name: Package FStar
+      run: |
+        source FStar/.github/env.sh
+        PACKAGE_DOCS=0 make -j -C FStar package
+    - name: Upload artifact
+      uses: actions/upload-artifact@v1
+      with:
+        name: fstar_package
+        path: FStar/src/ocaml-output/fstar_0.9.7.0-alpha1_Darwin_x86_64.tar.gz
```

Comment:
can we upload an unlimited number of artifacts? should this be the preferred way of distributing packages now, or is fstarlang/binaries still a good idea?

Reply:
I think so. It says "Storing artifacts uses storage space on GitHub. GitHub Actions usage is free for public repositories." [here](https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts).
I think having the binaries in some easy to find place would be nicer. These artifacts here are primarily to share build and test state between jobs and get build artifacts for testing/debugging I think. I also didn't find anything about their lifetime.
Another option (other than fstarlang/binaries) would be tagging releases in the repository itself.